From a0c7c44db7d3328c9e8b004a3df5a2d7418b06aa Mon Sep 17 00:00:00 2001
From: Hugo Osvaldo Barrera <hugo@whynothugo.nl>
Date: Mon, 8 Jun 2026 19:15:39 +0200
Subject: [PATCH 3/3] Bail if building the web UI fails

If building the web UI fails (e.g.: due to some network glitch), the
build process emits a warning and continues, producing binaries where
the web UI doesn't work.

Bail instead of producing broken binaries.
---
 scripts/ui-assets.cmake | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/scripts/ui-assets.cmake b/scripts/ui-assets.cmake
index 0c1c4de55..bab09b518 100644
--- a/scripts/ui-assets.cmake
+++ b/scripts/ui-assets.cmake
@@ -315,6 +315,10 @@ if(BUILD_UI)
     endif()
 endif()
 
+if(NOT provisioned)
+    message(FATAL_ERROR, "UI: npm failed, bailing.")
+endif()
+
 # ---------------------------------------------------------------------------
 # 3. Priority 3: HF Bucket download (if npm did not produce assets and HF_ENABLED=ON)
 # ---------------------------------------------------------------------------
-- 
2.55.0

